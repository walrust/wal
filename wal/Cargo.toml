[package]
name = "wal"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
wal-component = {path = "../wal-component"}
wal-vdom = {path = "../wal-vdom"}

[dependencies.web-sys]
version = "0.3.64"
features = [
    "console", 
    "Window",
    "Document",
    "Node",
    "Element",
    "Text",
]