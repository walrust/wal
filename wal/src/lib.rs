pub mod virtual_dom;
pub mod component;
pub mod app;
pub mod utils;