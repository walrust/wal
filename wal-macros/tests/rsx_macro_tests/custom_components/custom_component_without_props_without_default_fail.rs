use wal::{
    component::{behavior::Behavior, Component},
    virtual_dom::VNode,
};
use wal_macros::rsx;

include!("../utils/custom_components/custom_component_props_struct_without_default.rs");

fn main() {
    rsx! { <CustomComponentPropsStructWithoutDefault /> };
    rsx! { <CustomComponentPropsStructWithoutDefault key="key" />};
}
