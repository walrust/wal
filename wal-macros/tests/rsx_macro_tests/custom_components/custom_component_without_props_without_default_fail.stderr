error[E0277]: the trait bound `PropsWithoutDefault: Default` is not satisfied
  --> tests/rsx_macro_tests/custom_components/custom_component_without_props_without_default_fail.rs:10:14
   |
10 |     rsx! { <CustomComponentPropsStructWithoutDefault /> };
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `PropsWithoutDefault`
   |
help: consider annotating `PropsWithoutDefault` with `#[derive(Default)]`
  --> tests/rsx_macro_tests/custom_components/../utils/custom_components/custom_component_props_struct_without_default.rs
   |
   + #[derive(Default)]
   | };
   |

error[E0277]: the trait bound `PropsWithoutDefault: Default` is not satisfied
  --> tests/rsx_macro_tests/custom_components/custom_component_without_props_without_default_fail.rs:11:14
   |
11 |     rsx! { <CustomComponentPropsStructWithoutDefault key="key" />};
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `PropsWithoutDefault`
   |
help: consider annotating `PropsWithoutDefault` with `#[derive(Default)]`
  --> tests/rsx_macro_tests/custom_components/../utils/custom_components/custom_component_props_struct_without_default.rs
   |
   + #[derive(Default)]
   | };
   |
